flux/schnell/model/model.py

```diff
@@ -41,11 +47,35 @@ def convert_to_b64(self, image: Image) -> str:
         img_b64 = base64.b64encode(buffered.getvalue()).decode("utf-8")
         return img_b64
 
-    def predict(self, model_input):
+    def _generate_image(self, prompt, prompt2, guidance_scale, max_sequence_length,
+                        num_inference_steps, width, height, generator):
+        time.sleep(15)
+        try:
+            image = self.pipe(
+                prompt=prompt,
+                prompt_2=prompt2,
+                guidance_scale=guidance_scale,
+                max_sequence_length=max_sequence_length,
+                num_inference_steps=num_inference_steps,
+                width=width,
+                height=height,
+                output_type="pil",
+                generator=generator,
+            ).images[0]
+            self._result = image
+            return
+        except Exception as e:
+            logging.info(f"Image generation was aborted or failed: {e}")
+            self._exception = e
+            return
+
+    async def predict(self, model_input, request: fastapi.Request):
```

Comment:
I wonder if we can keep this sync, and use something like this to check for disconnection. ie:

```
from anyio import from_thread
def predict(self, model_input, request: fastapi.Request):
     is_disconnected = from_thread.run(request.is_disconnected)
     if is_disconnected:
         raise HTTPException(...)    
```

cancelling the ongoing inference is tricky though